### PR TITLE
feat: default id for connector label

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res [hunk 1]
```diff
@@ -205,7 +321,7 @@ module ConnectorConfigurationFields = {
           selectedConnector
         />
       }}
-      <RenderConnectorInputFields
+      <RenderConnectorLabel
```
Comment (jainlokesh318) [inline src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res:324]: Does this Component only render the label and no input/related info?

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res [hunk 2]
```diff
@@ -142,6 +201,63 @@ module RenderConnectorInputFields = {
...
+      ->LogicUtils.safeParse
+      ->FeatureFlagUtils.featureFlagType
+    let keys =
+      details->Js.Dict.keys->Js.Array2.filter(ele => !Js.Array2.includes(keysToIgnore, ele))
+    keys
+    ->Array.mapWithIndex((field, index) => {
+      let label = details->getString(field, "")
+      let formName = isLabelNested ? `${name}.${field}` : name
+      <UIUtils.RenderIf condition={label->Js.String2.length > 0}>
+        <div key={index->string_of_int}>
```
Comment (jainlokesh318) [inline src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res:232]: The label can be used as key of the React element

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res [hunk 3]
```diff
@@ -56,6 +58,63 @@ let inputField = (
...
+  let profileId =
+    ReactFinalForm.useField(`profile_id`).input.value->LogicUtils.getStringFromJson("")
+  let profileName = (
+    arrayOfBusinessProfile
+    ->Js.Array2.find((ele: HSwitchSettingTypes.profileEntity) => ele.profile_id === profileId)
+    ->Belt.Option.getWithDefault(defaultBusinessProfile)
+  ).profile_name
+  let connectorLabelOnChange = ReactFinalForm.useField(`connector_label`).input.onChange
+
+  React.useEffect1(() => {
```
Comment (jainlokesh318) [inline src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res:84]: Can't we set this default value on the time of initial generation itself

**Review by JeevaRamu0104 — APPROVED** [review]

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res:84]: We checked with that flow but it is not re-rendering as the flow we want and if we apply memos in that it was making the profile dropdown issue as they all are using same component.

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res:324]: yes only label

**Review by JeevaRamu0104 — APPROVED** [review]

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res (hunk 1 above)
Comment (jainlokesh318) [reply src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res:324]: But the code says it render the inputBox also?

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res]: Updated

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res:299]: Label is an inputField that's why it's an input field

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by JeevaRamu0104 — DISMISSED** [review]

File: src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res (hunk 1 above)
Comment (jainlokesh318) [reply src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res:263]: Now Is this RenderConnectorLabel required, or we can revert to old?

**Review by jainlokesh318 — APPROVED** [review]

**Pull request merged**